### DISCUSSION TITLE
TWO projection lines showing during roast

### DISCUSSION THREAD

**Discussion by hermetheuscoffee** [artisan-roaster-scope/artisan#963] [General]
I'm having an issue in Artisan (2.4.6) where TWO BT projection lines are displayed. One of these lines originates from the BT point (as expected), but the other projection line is always "anchored" to the BT at the charge point. Thus, these aren't just duplicate lines... they project different values, and only one of the two projection lines moves. 
This probably has something to do with my background profile, because it only happens for certain profiles that I use, but for the life of me, I can't figure out what the issue is. While I'm not in front of my roaster at the moment, I do have a screenshot I just now took that at least shows these two projection lines. (Note that one anchors to the charge temp and the other anchors to the current time/BT). Any ideas what could be causing this behavior? It might also be worth nothing that when loading this profile, the background RoR curve doesn't display unless I toggle OFF/ON. Only then will the background RoR appear. Any clues would be appreciated!

EDIT: In playing around with this, I have found another clue. Changing background profiles on-the-fly (which I do via buttons mapped to the Artisan command "loadBackground") "resets" the second projection line "anchor" to the point in time that the latest profile was loaded. So if I have background roast profiles named A and B, I can start the roast with A and the second projection line will be anchored to the 0:00 time. Change the background profile to B at 0:30 and the second projection line will now be anchored to the temperature at 0:30. Switch back to A at 1:00 and the second line will now be anchored to the time and temperature at the 1:00 mark. I hope this additional detail helps.
![artisan_dual_projection](https://user-images.githubusercontent.com/95500043/191998052-de6cb514-139f-4b4e-8997-705477c72e72.jpg)

**Comment by poundy** [comment]
While this is most likely going to be about your configuration, I'd first suggest going to 2.6 as the first step. There's no development in older versions, and there's no other reason to remain behind, so going forward will help ensure we minimise support efforts, as this is a donation supported project.

**Reply by poundy** [reply]
I also didn't see your additional details in the email notification...
Can you confirm though, if you switch to background profile C (one that never observes this) that it all goes away and corrects itself? 
It looks more and more like you will need to give us a copy of your configuration and the roast logs in question...

**Comment by MAKOMO** [comment]
The projections are based on the current roasts RoR signals and should be totally independent of any background profile. What you observe is very strange. Any reason you are not using v2.6.0?

Please attach your artisan settings (export via menu Help >> Save Settings) and those profiles as .alog files here to the discussion board using drag-and-drop (you might need to rename those files such that they carry a .txt extension to make this work). We will try to use those files to reproduce the issue and in the best case resolve it. 

Thanks for your support!

**Reply by hermetheuscoffee** [reply]
Thank you, poundy and Marko. The primary reason that I'm not on 2.6 is due to the changes made to the PID algorithm that don't play nicely with my PID setup (Yoctopuce-based). I'm going by memory, but I feel as if version 2.6 started feeding the PID algorithm a processed/smoothed/filtered temp reading instead of a raw reading. I feel like this was going to be roll back to the prior behavior in 2.6.1, but until it's rolled back I feel like 2.4.6 is my best option. Please correct if any of my statements are incorrect there.
[artisan-settingsPNG_Sumatra.zip](https://github.com/artisan-roaster-scope/artisan/files/9641094/artisan-settingsPNG_Sumatra.zip)
My attached Zip file can easily illustrate the two problems I'm having (and this same problem happens on all three of my roasting PCs, which are all Windows 10 machines). I've tried to make the steps as easy as possible to duplicate:
1. Extract all files to the root of your C drive 
2. Import the settings.
3. Load the "Sumatra" profile by using the BUTTON that I have configured for loading the Sumatra profile (it's configured to look in C:/ for the profile, so that's why step 1 has you extract there).
4. START and CHARGE the roast to begin graphing. Everything should look normal because Sumatra works just fine.
5. While the roast is active, click the PNG button. You'll now notice TWO odd behaviors. First, the RoR curve doesn't appear. Second, you'll now see two "projection" lines. 
6. You can toggle back and forth between Sumatra and PNG to see this behavior.

I'm sure it has something to do with SOMETHING that I've messed up in the profile, but I simply can't figure it out. I've used a file compare tool and have spent far too much time trying to find differences that could point to the culprit, but there's unfortunately a lot of info in the profile file that I don't understand.

Any advice you could give would be helpful. It's worth noting that MOST of my profile files behave like PNG, with the missing RoR and duplicate projections. I only have two profiles that seem to work correctly. Thank you!

**Reply by MAKOMO** [reply]
Thanks to your excellent description I could quickly reproduce the issue on 2.4.6 as well as 2.6.0. However, the current continuous 2.6.1 build seems to have fixed both of your issues. Note that this one also reverts the PID input to the one used in 2.4.6 and should thus work fine for you.  You can find those builds here: https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

**Reply by hermetheuscoffee** [reply]
Simply beautiful! I wasn't aware of the continuous build, but it fixed both issues and works great. I sent a donation to the project (and am also a paid Artisan Plus user and loving it). Thank you so much.
-Jason
